File: backend/schemas/requests.py
from datetime import datetime
from decimal import Decimal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field


class ItemRequestCreate(BaseModel):
    title: str = Field(min_length=3, max_length=140)
    category: str = Field(min_length=2, max_length=120)
    description: str = Field(min_length=10, max_length=3000)
    max_daily_budget: float = Field(ge=0)
    duration_days: int = Field(ge=1)
    location: str = Field(min_length=2, max_length=255)
    # Base64 data URL (data:image/...;base64,...) — opsiyonel temsili görsel
    image_base64: str | None = None


class RequesterPreview(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    name: str | None = None
    avatar_base64: str | None = None


class ItemRequestRead(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: UUID
    requester_id: UUID
    title: str
    category: str
    description: str
    max_daily_budget: Decimal
    duration_days: int
    location: str
    image_base64: str | None = None
    status: str
    created_at: datetime
    requester: RequesterPreview | None = None

File: frontend/types/requests.ts
export interface RequesterPreview {
  id: string;
  name: string | null;
  avatar_base64: string | null;
}

export interface ItemRequest {
  id: string;
  requester_id: string;
  title: string;
  category: string;
  description: string;
  max_daily_budget: number;
  duration_days: number;
  location: string;
  image_base64?: string | null;
  status: string;
  created_at: string;
  requester?: RequesterPreview | null;
}

export interface ItemRequestCreatePayload {
  title: string;
  category: string;
  description: string;
  max_daily_budget: number;
  duration_days: number;
  location: string;
  image_base64?: string | null;
}

export interface Offer {
  id: string;
  item_request_id: string;
  supplier_id: string;
  price_amount: number;
  description: string;
  status: string;
  created_at: string;
}

export interface RentalDeal {
  id: string;
  item_request_id: string;
  accepted_offer_id: string;
  escrow_status: string;
  delivery_confirmed_at: string | null;
  deal_status: string;
  created_at: string;
}

export interface DealSummary {
  id: string;
  item_request_title: string;
  offer_price: number;
  escrow_status: string;
  deal_status: string;
  role: "requester" | "supplier";
  created_at: string;
}

export interface MessageSender {
  id: string;
  name: string | null;
  avatar_base64: string | null;
}

export interface DealMessage {
  id: string;
  thread_id: string;
  sender_id: string;
  body: string;
  created_at: string;
  sender?: MessageSender | null;
}

export interface HomeItemRequest {
  id: string;
  title: string;
  imageUrl?: string | null;
  maxDailyBudget: number;
  durationDays: number;
  location: string;
}
